wikimedia/wikipedia-ios
# On This Day widget tweaks

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T262876

### Notes
UI changes:
- Placeholder widget is now localized
- The largest dynamic type respected by widgets (the "even larger" sizes provided in an accessibility setting do not make it larger than `extraExtraLarge`) was cutting off some text. It no longer is. (@carolynlimadeo: If instead of tightening it up, we wanted to drop the article box for these sizes - allowing a more-useful amount of event snippet text - let me know.)

Code improvements:
- Force unwraps removed (and made the articleURL optional in the process)
- Removed some placeholder text
- Use a cached version of the article image when fetching it for a preview.

### Test Steps
1. Try the widget with the largest dynamic type size.
1. Easiest way to test placeholder entry text: Replace `completion(entry)` in `OnThisDayWidget.swift` with `completion(self.placeholderEntry)`. Look at the widget in, say, English and Arabic, make sure both look appropriate. (Some text is yet to be translated in non-English languages, so you'll likely still see some English.)

### Screenshots/Videos
**Before**
Second largest size (looks fine)
<img width="150" alt="Screenshot" src="https://user-images.githubusercontent.com/9295855/93652675-c01c1f80-f9ca-11ea-87ac-619abe47efbe.png">
 
Largest size (broken)
<img width="150" alt="Screenshot 1" src="https://user-images.githubusercontent.com/9295855/93652700-d0cc9580-f9ca-11ea-99e3-838da170bab2.png">

**After**
Largest size (fixed w/ slightly tighter vertical spacing in a few spots)
<img width="150" alt="Screenshot 2" src="https://user-images.githubusercontent.com/9295855/93652706-d88c3a00-f9ca-11ea-9596-768d74dbf97a.png">
